app/services/fact_importer.rb

```diff
@@ -55,17 +57,16 @@ def fact_name_class
   attr_reader :host, :facts
 
   def delete_removed_facts
-    to_delete = host.fact_values.eager_load(:fact_name).where("fact_names.type = '#{fact_name_class}' AND fact_names.name NOT IN (?)", facts.keys)
-    # N+1 DELETE SQL, but this would allow us to use callbacks (e.g. auditing) when deleting.
-    deleted = to_delete.destroy_all
-    @counters[:deleted] = deleted.size
-
-    @db_facts = nil
-    logger.debug("Merging facts for '#{host}': deleted #{counters[:deleted]} facts")
+    # deletes all facts using a single SQL query (with inner query)
+    deleted_count = FactValue.joins(:fact_name).where(:host => host, 'fact_names.type' => fact_name_class).where.not('fact_names.name' => facts.keys).delete_all
+    @counters[:deleted] = deleted_count
+    ActiveSupport::Notifications.instrument "fact_importer_deleted.foreman", :host_id => host.id, :host_name => host.name, :facts => facts, :count => deleted_count do
+      logger.debug("Merging facts for '#{host}': deleted #{deleted_count} facts")
```

Comment:
why do we log here the same thing that is logged by the listener but with a different level?